### DISCUSSION TITLE
Estimativas

### DISCUSSION THREAD

**Discussion by jardelsantana** [frontendbr/forum#780] [Perguntas e Respostas]
### Faaaaaaala galera!
O que vocês costumam utilizar para definir o tempo de entrega dos seus projetos? Na faculdade aprendemos a utilizar a APF ([Análise de Pontos de Função](https://pt.wikipedia.org/wiki/Análise_de_pontos_de_função)) para definir o tempo de entrega de um projeto... E na **vida real**, usamos o quê? Usam algum software para medir isso?

**Comment by kvnol** [comment]
Eu sempre fiz no achismo, mas sei que é bem bosta. Estou acompanhando aqui.

**Comment by klarkc** [comment]
No Scrum você usa uma medida subjetiva que pode obedecer a qualquer regra e limite, porém o mais importante é que essa estimativa sofra ajustes durante o processo, o próprio scrum facilita isso em sua metodologia. No início as estimativas são bem imprecisas, mas com o tempo e com a retro-alimentação de dados, vão passando a ser mais assertivas. Nas reuniões de refinamento do backlog o time discute o tamanho das tarefas, de forma que essas estimativas se tornem cada vez mais precisas.

Lembrando que nada impede de medir horas no scrum, para fins de cálculo de custo. O que é desincentivado é o uso de horas como base de estimativas, nesse caso as horas levam em consideração a velocidade do time e o tamanho do projeto, e talvez mais algum cálculo de flutuação de velocidade. Mas nunca apenas horas imaginadas. A partir disso você pode levantar um gráfico de gantt com entrega de épicos x releases, e mirar nessas entregas, sempre em acordo com o time.

**Comment by fernandofleury** [comment]
Geralmente você estima pela suposta dificuldade. é suposto porque você não tem com confirmar de fato até você chegar e fazer. existem mecanismos dentro das metodologia que ajudam nessa assertividade, tipo os critérios de pronto da tarefa (ter layout já, ter API ou não, possíveis problemas)

Mas o que facilita de verdade o processo é ter uma tarefa pivô pra isso. No final das contas tanto faz como você vai querer fazer a medição, ela sempre vai ter bons valores por causa do pivô que foi definido. Por exemplo:

Você tem 3 tarefas a serem estimadas:

- Adicionar cor ao botão
- Adicionar texto na página de sucesso
- Adicionar alerta no erro da requisição da compra

Geralmente você pega uma das tarefas e seleciona como pivô (se você já estiver num processo de sprint você pode usar uma tarefa do sprint passado caso você ache que a medição foi assertiva). Ai você estima essa tarefa, vamos usar o **Adicionar texto na página de sucesso**

Aí vai depender de qual sistema de pontos que você tá usando, mas vamos usar números inteiro e já era.

Numa escala de 0 a 10 podemos considerar que adicionar texto é tipo um 2, afinal não é super complexo e tem baixa dificuldade no geral. (Lembrando que o número é completamente relativo, vocês poderiam medir com frutas se quisessem)

Feito isso podemos estimar as outras tarefas:

- Adicionar cor no botão (Considerando que é mais fácil que adicionar o texto, já que é só CSS) 1 ponto
- Adicionar alerta na requisição (Considerando que é mais complexo) 5 pontos

Ora, mas de onde tiramos 5, ou 3 ou 1 ou assim vai? Na real esses números realmente não importam. O que importa é você conseguir gerar relação entre eles, e o valor final deles pode ser qualquer um (não existe inflação de velocity, o que existe só é velocity).

Iniciado esse processo de medição, sua velocity se baseia nesses valores, então se vocês acham que a pontuação está muito alta (estamos dando 80 pra uma tarefa e parece absurdo) é só reduzir o valor bruto das operações. Geralmente usamos fibonnaci pras medições.

Espero que esse textão faça sentido, mas é que na real é uma parada muito mais de feeling e correção com o tempo do que uma ciência exata.

**Comment by klarkc** [comment]
Uso também fibonacci, ele é muito bom porque 'ignora proporção', não existem dobros, triplos, etc. é um numero bem exponencial e a diferença entre valores vizinhos é bem delineada.
